From: Sergio Gelato <Sergio.Gelato@astro.su.se>
Date: 2014-09-01
Subject: nfsd: Fix ACL null pointer deref
Bug-Debian: https://bugs.debian.org/754420
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1348670
Origin: https://bugs.launchpad.net/debian/+source/linux/+bug/1348670/+attachment/4192076/+files/nfsd-fix-acl-null-pointer-deref.patch

Fix regression introduced in 3.2.60 by cherry-picking a post-3.14 patch that
depends on the set_acl methods being able to cope with a NULL ACL argument.

See Debian #754420, LP#1348670.
--- a/fs/nfsd/vfs.c
+++ b/fs/nfsd/vfs.c
@@ -508,6 +508,9 @@
 	char *buf = NULL;
 	int error = 0;
 
+	if (!pacl)
+		return vfs_setxattr(dentry, key, NULL, 0, 0);
+
 	buflen = posix_acl_xattr_size(pacl->a_count);
 	buf = kmalloc(buflen, GFP_KERNEL);
 	error = -ENOMEM;
